src/main/java/org/junit/internal/runners/JUnit38ClassRunner.java

```diff
@@ -170,6 +174,13 @@ public void sort(Sorter sorter) {
         }
     }
 
+    public void order(GenericOrdering ordering) throws InvalidOrderingException {
```

Comment:
Why the duplication?  I can see that one would want to be able to do both @OrderWith(MyOrdering.class) and @OrderWith(MySorter.class), and that's difficult unless one of the classes extends the other.  But it looks as though Ordering is essentially a marker interface at this point: most of the code that actually does anything is assuming that all Orderings are either Sorters or GenericOrderings.  If you _like_ the RunRule suggestion below, this becomes more of an asset than a liability, but if not, maybe there's some further shuffling that could happen here.